File: sText/src/main/java/com/samid/sTextView/base/ISText.java
package com.samid.sTextView.base;

import android.graphics.Canvas;
import android.util.AttributeSet;

public interface ISText {
    void init(STextView hTextView, AttributeSet attrs, int defStyle);

    void animateText(CharSequence text);

    void onDraw(Canvas canvas);

    void setAnimationListener(AnimationListener listener);
}

File: sText/src/main/java/com/samid/sTextView/base/ISText.kt
package com.samid.stextview.base

import android.graphics.Canvas
import android.util.AttributeSet

interface ISText {
    fun init(sTextView: STextView, attrs: AttributeSet?, defStyle: Int)
    fun animateText(text: CharSequence)
    fun onDraw(canvas: Canvas)
    fun setAnimationListener(listener: AnimationListener)
}